### PR TITLE
test: add testing of new file operations

### LINKED ISSUE #9131: Add basic tests for `FilePath` and friends
We don't need tests for everything, but I would like 5-8 tests like:

```
@Test
def exists01(): Unit \ IO = 
  FilePath.handle(() -> FilePath.exists("foo.txt"))()
```

or something like that.

### PR DESCRIPTION
**Pull request by sockmaster27**
Fixes  #9131

### CODE REVIEW

**Comment by magnus-madsen** [comment]
I think this is good enough now. Later we can return to test every operator, but for now we just want to ensure that the whole setup with effect handlers works.

File: main/test/ca/uwaterloo/flix/library/TestFileRead.flix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+mod TestFileRead {
+
+    import java.io.IOException
+    import java.nio.file.Path
```
Comment (magnus-madsen) [inline main/test/ca/uwaterloo/flix/library/TestFileRead.flix:20]: These are not needed, right?

**Comment by sockmaster27** [comment]
> I think this is good enough now. Later we can return to test every operator, but for now we just want to ensure that the whole setup with effect handlers works.

Ah, ok. But we can still do one for each `FileRead`, `FileReadWithResult`, `FileWrite` and `FileWriteWithResult`, right?

**Comment by magnus-madsen** [comment]
> > I think this is good enough now. Later we can return to test every operator, but for now we just want to ensure that the whole setup with effect handlers works.
> 
> Ah, ok. But we can still do one for each `FileRead`, `FileReadWithResult`, `FileWrite` and `FileWriteWithResult`, right?

Lets just do FileReadWithResult

**Review by magnus-madsen — APPROVED** [review]

**Comment by magnus-madsen** [comment]
@sockmaster27 test failures

**Comment by sockmaster27** [comment]
@magnus-madsen 
Looks like the working directory is different for IntelliJ tests and CI tests.
Is it ok if we leave out the test case of finding an existing file? Then it is ready.

**Comment by magnus-madsen** [comment]
I don't care too much about the invalid path. Lets just check that an existing file exists and can be read?

**Comment by sockmaster27** [comment]
@magnus-madsen RFR

File: main/test/ca/uwaterloo/flix/library/TestFileRead.flix [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+
+    /////////////////////////////////////////////////////////////////////////////
+    // exists                                                                  //
+    /////////////////////////////////////////////////////////////////////////////
+
+    @Test
+    def exists01(): Bool \ IO = {
+        (
+            makeTempPath("some-file") >>= path ->
```
Comment (magnus-madsen) [inline main/test/ca/uwaterloo/flix/library/TestFileRead.flix:55]: Lets make this some name that is definitely unique. I propose:

`flix-tmpfile-68975`

(I am just scared overwriting data)

File: main/test/ca/uwaterloo/flix/library/TestFileRead.flix [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    use Monad.{>>=}
+
+    import java.io.IOException
+    import java.nio.file.Path
+
+    ///
+    /// Helper - make a filepath within the temp directory.
+    /// Use `java.nio.file.Path` functions for platform independence.
+    ///
+    def makeTempPath(fileName: String): Result[IoError, String] = region rc {
```
Comment (magnus-madsen) [inline main/test/ca/uwaterloo/flix/library/TestFileRead.flix:30]: Move this function after the tests

Comment (magnus-madsen) [reply main/test/ca/uwaterloo/flix/library/TestFileRead.flix:30]: Do you think this should be a AE? Part of FileWrite?

Comment (magnus-madsen) [reply main/test/ca/uwaterloo/flix/library/TestFileRead.flix]: AE = algebraic effect, i.e. part of FileWrite.

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**